genet/core.py

```diff
@@ -1554,6 +1554,7 @@ def route_schedule(self, services: Union[list, set] = None, solver='cbc', allow_
                         service_g = service.graph()
 
                         for route_group, graph_group in zip(routes, graph_groups):
+                            route_group = list(route_group)
```

Comment:
Oops, missed this! For consistency, either change this to what I did in the similar case later on in the file (lines 1650 and 1652) or change what I did to match what you've done. I have no opinion either way.